src/mongocxx/test/spec/monitoring.cpp

```diff
@@ -28,63 +28,39 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace spec {
 
 using namespace mongocxx;
+using bsoncxx::to_json;
 
 void apm_checker::compare(bsoncxx::array::view expectations,
                           bool allow_extra,
                           const test_util::match_visitor& match_visitor) {
-    using bsoncxx::to_json;
+    auto is_kill_cursor = [](bsoncxx::document::value v) {
+        return v.view()["command_started_event"]["command"]["killCursors"];
+    };
 
     auto events_iter = _events.begin();
+    auto events_end = (_skip_kill_cursors)
+                          ? std::remove_if(_events.begin(), _events.end(), is_kill_cursor)
+                          : _events.end();
+
```

Comment:
"Removing" the `killCursors` early on simplified most of this logic. At first, `killCursor` commands were interwoven in the list of events. Some of the new tests would try to match one of these `killCursor` commands with an `expectation` in the loop below. I think it's simpler to just move these commands out of the way if we don't want to use them rather than adding checks in our loops for them.

Reply:
Good idea. Done